docs/src/main/asciidoc/security-oidc-bearer-token-authentication.adoc

```diff
@@ -399,6 +401,53 @@ For example, if you work with Keycloak, you can use `keycloak.js` to authenticat
 </html>
 ----
 
+==== Use OIDC proxy
+
+Instead of tying your SPA to a specific OIDC provider by this OIDC provider's script, you may want to consider implementing an authorization code flow directly in the SPA by redirecting the users to a configurable OIDC authorization endpoint, accepting the OIDC redirect in the SPA and completing the authorization code flow by sending the code to a standard OIDC token endpoint and finaly sending the acquired access tokens to a Quarkus endpoint.
+
+Quarkus introduces an experimental `quarkus-oidc-proxy` OIDC proxy extension which provides an authorization code flow support for Quarkus OIDC `service` applications and can help to create SPAs by encapsulating all the OIDC provider configuration properties.
+
+All you need to do is to add the `quarkus-oidc-proxy` dependency to your Quarkus OIDC `service` application:
+
+[source,xml,role="primary asciidoc-tabs-target-sync-cli asciidoc-tabs-target-sync-maven"]
+.pom.xml
+----
+<dependency>
+    <groupId>io.quarkus</groupId>
+    <artifactId>quarkus-oidc-proxy</artifactId>
+</dependency>
+----
+
+[source,gradle,role="secondary asciidoc-tabs-target-sync-gradle"]
+.build.gradle
+----
+implementation("io.quarkus:quarkus-oidc-proxy")
+----
+
+Now, the SPA can use the OIDC proxy endpoints for authenticating the users using the authorization code flow and send the acquired access tokens to the Quarkus OIDC `service` application.
+
+For example, if the endpoint is listening at `http://localhost:8080`, then, after adding this dependency, the OIDC authorization endpoint is available at `http://localhost:8080/oidc/authorize` , the OIDC token endpoint is available at `http://localhost:8080/token`, the OIDC JWKS endpoint is available at `http://localhost:8080/oidc/jwks` and finally, the OIDC well known configuration endpoint which supports the discovery is available at `http://localhost:8080/.well-known/openid-configuration`. These endpoint paths can be customized.
```

Comment:
So, why does everything live under `/oidc` except `/token`? Judging by the config file, this is a documentation bug.

For Renarde I have some generated endpoints as well, and I prefixed them all under `/_renarde`. Perhaps in this case the default should be `/q/oidc`?

Reply:
@FroMage I'll fix it, I'm not sure putting it under `/q` which a reserved path, `/oidc/*` is done by default but users can change to whatever they prefer

Reply:
@FroMage Yeah, I've seen WebAuthn uses `/q`. The problem with the OIDC proxy could be that `/q` becomes visible part of the OIDC proxy endpoints, and we already have users asking to hide various `q` related hints that it is Quarkus - this is easy to fix by re-configuring a non-application path value but it affects all `/q` endpoint, while having some root path for the OIDC proxy which has some OIDC related word seems reasonable.

FYI the OIDC proxy root path is configurable, so if someone will prefer to align with `/q` they will set the OIDC proxy property and it is done. I'm just not sure about doing it by default

Reply:
Well, yeah, but the default path should be something we discuss, probably best to send an email to quarkus-dev.